pkg/reconciliation/secrets.go

```diff
@@ -192,8 +192,13 @@ func (rc *ReconciliationContext) keystoreCASecret() types.NamespacedName {
 
 func (rc *ReconciliationContext) retrieveInternodeCredentialSecretOrCreateDefault() (*corev1.Secret, error) {
 	secret, retrieveErr := rc.retrieveSecret(rc.keystoreCASecret())
-	if retrieveErr != nil {
-		if errors.IsNotFound(retrieveErr) {
+	_, retrieveBootStrappingErr := rc.retrieveSecret(types.NamespacedName{
+		Name:      fmt.Sprintf("%s-keystore", rc.Datacenter.Name),
+		Namespace: rc.Datacenter.Namespace,
+	})
+	if retrieveErr != nil || retrieveBootStrappingErr != nil {
```

Comment:
This if clause jungle is a bit difficult to follow. Wouldn't it make more sense to check one by one and continue? Such as:

```go
secret, err := rc.retrieveSecret(rc.keystoreCASecret())
if err != nil {
  if  errors.IsNotFound(err) { // Handle here }
  return nil, err
}

// The same checks for retrieveBootstrapping
```

That would avoid the original problem. Also, the ``rc.createCABootstrappingSecret`` logic is now repeated in two different conditional clauses and that could cause issues in some future refactoring and it duplicates the code.

Reply:
Your suggestion makes more sense. I refactored the code a bit, can you check if it looks better to you?